### PR TITLE
Update ManiphestExcelDefaultIncludeCustomFieldsFormat.php

### PR DESCRIPTION
**Pull request by stevexyz**
Added points + custom fields of type date formatted as date + solved "break" issue in columns management

### CODE REVIEW

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php [hunk 1]
```diff
@@ -173,34 +173,33 @@ public function buildWorkbook(
...
-             ->execute();
-           $ppositions = mpull($ppositions, null, 'getObjectPHID');
-           foreach ($ppositions as $pposition) {
-             if ($pposition->getSequence() == 0) {
-               continue;
-             }
-             $pposition->attachColumn($task_column);
-             if (empty($task_to_column[$task_id])) {
-               $task_id_to_column[$task_id] = array();
+      if (count($columns) > 0) {
```
Comment (neandrake) [inline extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:176]: What reason to change this? I usually prefer to break/return early and save indentation. From what I can tell nothing here changed functionally.

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php [hunk 2]
```diff
@@ -25,35 +24,35 @@
  *   does seem to have some methods for setting/requiring a "viewer".
  *   This was not needed in the moment so it's ignored for the time being.
  */
-
-final class ManiphestExcelDefaultIncludeCustomFieldsFormat extends ManiphestExcelFormat {
+final class ManiphestExcelDefaultIncludeCustomFieldsFormatExtended extends ManiphestExcelFormat {
   public function getName() {
-    return pht('Default with Custom Fields');
+    return pht('Default with Custom Fields - extended');
```
Comment (neandrake) [inline extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:29]: If I merge this in I don't think there's any reason to have a different name

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php [hunk 3]
```diff
@@ -25,35 +24,35 @@
...
     // title => the displayed header title in the spreadsheet, in row 0
     // width => initial width in pixels for the column, null leaves unspecified
     // celltype => which format the column data should be set as, default is STRING
     //   can be null if it's a date field
     // isDate => there is no date format in the PHPExcel_Cell_DataType, so this is needed
     // cftype => the custom field data type, only specified for custom field headers
-
     $colHeaders = array(
       array(
+        'title' => pht('URI'),
```
Comment (neandrake) [inline extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:50]: I would prefer to keep ID as the first column

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php [hunk 4]
```diff
@@ -124,13 +122,14 @@ public function buildWorkbook(
       // getFieldType() method which is needed here.
       $fieldName = $customField->getProxy()->getFieldName();
       $fieldType = $customField->getProxy()->getFieldType();
-
       $isDateField = $fieldType == 'date';
-      $cellType = PHPExcel_Cell_DataType::TYPE_STRING;
-      if ($fieldType == 'int') {
+      if ($isDateField) {
+        $cellType = null;
```
Comment (neandrake) [inline extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:127]: Is there a a `PHPExcel_Cell_DataType::TYPE_DATE`?

**Review by neandrake — CHANGES_REQUESTED** [review]
Thank you for the submission, I think this is a good change! In general I prefer the formatting I have currently as the new lines help me visually to logically separate "actions" happening. Would you be willing to reduce the change to just the addition of the extra column and date handling?

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php (hunk 2 above)
Comment (stevexyz) [reply extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:29]: Sure! In reality in my installation I kept the original and added this as new file (and for this reason I changed name)

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php (hunk 1 above)
Comment (stevexyz) [reply extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:176]: The break instruction (that I deleted) exits from loops, but there was no loop to exit from, and infact before removal I got a case with error that showed me that I have to correct
(PS: here in the descriptions I see red lines I didn't delete, do not know why, maybe just visualization problems)

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php (hunk 3 above)
Comment (stevexyz) [reply extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:50]: I put first and later even made column thinner (almost disappearing). Personally I would remove (Txxx is enough as reference if needed). Depends on you needs and taste ! :)

File: extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php (hunk 4 above)
Comment (stevexyz) [reply extensions/ManiphestExcelDefaultIncludeCustomFieldsFormat.php:127]: Seems not. Anyway I did coherently as in the other date fields present before!

**Comment by stevexyz** [comment]
You can cherry pick the changes. Spacing removals might depend on the copy/paste I did on web i/f, I also prefer spaced! :)

**Pull request closed without merging**